core/libraries/batch/JobHandlers/ExecuteBatchDeletion.php

```diff
@@ -98,37 +107,42 @@ public function continue_job(JobParameters $job_parameters, $batch_size = 50)
-                $ids_to_delete_this_query = array_slice($ids_to_delete, 0, $batch_size - $units_processed, true);
-                if ($model->has_primary_key_field()) {
-                    $where_conditions = [
-                        $model->primary_key_name() => [
-                            'IN',
-                            $ids_to_delete_this_query
-                        ]
-                    ];
+                if ($model instanceof EEM_Registration) {
+                    // Now THIS could be a command!
+                    $units_processed += $this->deleteRegistrationsAndTransactions($ids_to_delete, $batch_size - $units_processed, $job_parameters);
                 } else {
-                    $where_conditions = [
-                        'OR' => []
-                    ];
-                    foreach ($ids_to_delete_this_query as $index_primary_key_string) {
-                        $keys_n_values = $model->parse_index_primary_key_string($index_primary_key_string);
-                        $where_conditions['OR'][ 'AND*' . $index_primary_key_string ] = $keys_n_values;
+                    $ids_to_delete_this_query = array_slice($ids_to_delete, 0, $batch_size - $units_processed, true);
+                    if ($model->has_primary_key_field()) {
+                        $where_conditions = [
+                            $model->primary_key_name() => [
+                                'IN',
+                                $ids_to_delete_this_query
+                            ]
+                        ];
+                    } else {
+                        $where_conditions = [
+                            'OR' => []
+                        ];
+                        foreach ($ids_to_delete_this_query as $index_primary_key_string) {
+                            $keys_n_values = $model->parse_index_primary_key_string($index_primary_key_string);
+                            $where_conditions['OR'][ 'AND*' . $index_primary_key_string ] = $keys_n_values;
+                        }
+                    }
+                    // Deleting time!
+                    // The model's deletion method reports every ROW deleted, and in the case of CPT models that will be
+                    // two rows deleted for event CPT item. So don't rely on it for the count of items deleted.
+                    $model->delete_permanently(
+                        [
+                            $where_conditions
+                        ],
+                        false
+                    );
+                    $units_processed += count($ids_to_delete_this_query);
+                    $remaining_ids = array_diff_key($ids_to_delete, $ids_to_delete_this_query);
+                    // If there's any more from this model, we'll do them next time.
+                    if (count($remaining_ids) > 0) {
+                        $models_and_ids_remaining[ $model_name ] = $remaining_ids;
                     }
```

Comment:
would increase readability if this was extracted into a new method and then simply called here in this `else` block
but that can be done later